heat/core/manipulations.py

```diff
@@ -1942,101 +1942,111 @@ def shape(a: DNDarray) -> Tuple[int, ...]:
+def __pivot_sorting(
+    a: DNDarray, sort_op: Callable, axis: Optional[int] = None, descending: bool = False, **kwargs
+) -> Union[torch.Tensor, Tuple[torch.Tensor, torch.Tensor]]:
     """
-    Sorts the elements of `a` along the given dimension (by default in ascending order) by their value.
-    The sorting is not stable which means that equal elements in the result may have a different ordering than in the
-    original array.
-    Sorting where `axis==a.split` needs a lot of communication between the processes of MPI.
-    Returns a tuple `(values, indices)` with the sorted local results and the indices of the elements in the original data
+    Parallel sorting function for :func:`sort` and :func:`unique`, based on [1].
 
     Parameters
     ----------
     a : DNDarray
-        Input array to be sorted.
-    axis : int, optional
-        The dimension to sort along.
-        Default is the last axis.
-    descending : bool, optional
-        If set to `True`, values are sorted in descending order.
-    out : DNDarray, optional
-        A location in which to store the results. If provided, it must have a broadcastable shape. If not provided
-        or set to `None`, a fresh array is allocated.
-
-    Raises
-    ------
-    ValueError
-        If `axis` is not consistent with the available dimensions.
+        Distributed input data
+    axis : int or None
+        Axis along which the operation will be performed.
+    sort_op : torch operation
+        torch.sort or torch.unique
+    descending : bool
+        Whether :func:`sort` will return elements sorted in descending order. Default: `False`.
 
-    Examples
-    --------
-    >>> x = ht.array([[4, 1], [2, 3]], split=0)
-    >>> x.shape
-    (1, 2)
-    (1, 2)
-    >>> y = ht.sort(x, axis=0)
-    >>> y
-    (array([[2, 1]], array([[1, 0]]))
-    (array([[4, 3]], array([[0, 1]]))
-    >>> ht.sort(x, descending=True)
-    (array([[4, 1]], array([[0, 1]]))
-    (array([[3, 2]], array([[1, 0]]))
+    References
+    ----------
+    [1] Li et al., 1993, "On the versatility of parallel sorting by regular sampling", Parallel Computing, Volume 19, Issue 10, pages 1079-1103
     """
-    stride_tricks.sanitize_axis(a.shape, axis)
-
-    if a.split is None or axis != a.split:
-        # sorting is not affected by split -> we can just sort along the axis
-        final_result, final_indices = torch.sort(a.larray, dim=axis, descending=descending)
-
-    else:
-        # sorting is affected by split, processes need to communicate results
-        # transpose so we can work along the 0 axis
-        transposed = a.larray.transpose(axis, 0)
-        local_sorted, local_indices = torch.sort(transposed, dim=0, descending=descending)
-
-        size = a.comm.Get_size()
-        rank = a.comm.Get_rank()
+    size = a.comm.Get_size()
+    rank = a.comm.Get_rank()
+    transposed = a.larray.transpose(axis, 0)
```

Comment:
> this assumes a 2D torch tensor i believe. is this the case at this point?

No it doesn't, `torch.transpose` works like `np.swapaxes`, this will just transpose axis 0 with axis `axis` and leave the rest be.